addons/web/static/src/legacy/js/core/menu.js

```diff
@@ -59,11 +63,24 @@ export async function initAutoMoreMenu(el, options) {
         _adapt();
     }
 
-    const debouncedAdapt = () => {
-        clearTimeout(debounce);
-        debounce = setTimeout(_adapt, 250);
+    let newCall = true;
+    let refreshId = null;
+    // This should throttle the `_adapt()` method to the browser's refresh
+    // rate. The first menu adaptation is always executed immediately.
+    const throttleAdapt = () => {
+        const onRefresh = !newCall ? _adapt : () => {};
+        if (newCall) {
+            _adapt();
+            newCall = false;
+        }
+        window.cancelAnimationFrame(refreshId);
+        refreshId = window.requestAnimationFrame(() => {
+            onRefresh();
+            newCall = true;
+        });
```

Comment:
I am not sure about this implementation, it looks more like a debounce with a very small delay (the browser refresh rate) than a throttle to me. I am also wondering if you are not calling adapt too much, between refresh rates:
![image](https://github.com/odoo/odoo/assets/10338094/3ae18ed2-48d5-4fda-8f5f-6e94b6905f53)

But maybe I am overthinking. I am not sure how to implement it better, I am just wondering. Did you copy this implementation from somewhere? It does not seem to be the same as the `throttleForAnimation` one.